src/act-compat.js

```diff
@@ -19,8 +34,28 @@ function actPolyfill(cb) {
 
 const act = reactAct || actPolyfill
 
-function rtlAct(...args) {
-  return act(...args)
+// this will not avoid warnings that react-dom 16.8.0 logs for triggering
+// state updates asynchronously, but at least we can tell people they need
+// to upgrade to avoid the warnings.
+async function asyncActPolyfill(cb) {
+  if (actSupported) {
+    // if act is supported and async act isn't and they're trying to use async
+    // act, then they need to upgrade from 16.8 to 16.9.
+    // This is a seemless upgrade, so we'll add a warning
+    console.error(
+      `It looks like you're using a version of react-dom that supports the "act" function, but not an awaitable version of "act" which you will need. Please upgrade to at least react-dom@16.9.0 to remove this warning.`,
+    )
+  }
+  // make all effects resolve before
+  act(() => {})
```

Comment:
we went back and forth on this, and decided not to flush in the beginning. is there a reason you added this here?

Reply:
Thanks for that! I wasn't sure whether this was correct 😅 I'll remove that.